File: ssh/src/service/interf/UserService.java
package service.interf;

import java.util.List;

import model.User;

public interface UserService
{
	public User getById(Integer id);

	public void save(User entity);

	public void update(User entity);

	public void delete(User entity);

	public void deleteById(Integer id);

	public List<User> getAll();

	public User getByUserName(String userName);
}

File: ash/ash3/Service/Impl/UserService.cs
using System;
using System.Linq;
using System.Web;
using model;
using Spring.Transaction.Interceptor;
using System.Collections.Generic;
using ash3.Service;
using ash3.Dao.Hibernate;

namespace ash3.Service.Impl
{
    [Transaction]
    public class UserService : IUserService
    {
        public UserDao userDao { set; get; }

        [Transaction(ReadOnly = true)]
        public User getById(long? id)
        {
            return userDao.getById(id);
        }

        [Transaction]
        public void save(User model)
        {
            userDao.save(model);
        }

        [Transaction]
        public void update(User model)
        {
            userDao.update(model);
        }

        [Transaction]
        public void delete(User model)
        {
            userDao.delete(model);
        }

        [Transaction(ReadOnly = true)]
        public IList<User> getAll()
        {
            return userDao.getAll();
        }
    }
}
